client/components/command-palette/use-command-palette.tsx

```diff
@@ -74,73 +75,94 @@ const siteToAction =
 export const useCommandPalette = ( {
 	selectedCommandName,
 	setSelectedCommandName,
+	search,
 }: useCommandPaletteOptions ): { commands: Command[] } => {
 	const { data: allSites = [] } = useSiteExcerptsQuery(
 		[],
 		( site ) => ! site.options?.is_domain_only
 	);
 
-	// Sort sites in the nested commands to be consistent with site switcher and /sites page
 	const { sitesSorting } = useSitesSorting();
 	const sortedSites = useSitesListSorting( allSites, sitesSorting );
-
-	// Get current site ID to rank it to the top of the sites list
 	const { currentSiteId } = useCurrentSiteRankTop();
-
-	// Call the generateCommandsArray function to get the commands array
-	const commands = useCommandsArrayWpcom( {
-		setSelectedCommandName,
-	} );
-
+	const commands = useCommandsArrayWpcom( { setSelectedCommandName } );
 	const currentPath = useSelector( ( state: object ) => getCurrentRoute( state ) );
 
-	// Filter out commands that have context
-	const commandHasContext = ( paths: string[] = [] ): boolean =>
-		paths.some( ( path ) => currentPath.includes( path ) ) ?? false;
-
-	// Find and store the "viewMySites" command
-	const viewMySitesCommand = commands.find( ( command ) => command.name === 'viewMySites' );
-
-	// Sort the commands with the contextual commands ranking higher than general in a given context
-	const sortedCommands = commands
-		.filter( ( command ) => ! ( command === viewMySitesCommand ) )
-		.sort( ( a, b ) => {
-			const hasContextCommand = commandHasContext( a.context );
-			const hasNoContext = commandHasContext( b.context );
-
-			if ( hasContextCommand && ! hasNoContext ) {
-				return -1; // commands with context come first if there is a context match
-			} else if ( ! hasContextCommand && hasNoContext ) {
-				return 1; // commands without context set
+	// Logic for selected command (sites)
+	if ( selectedCommandName ) {
+		const selectedCommand = commands.find( ( c ) => c.name === selectedCommandName );
+		let sitesToPick = null;
```

Comment:
Thanks for the refactor!
Now it's more clear that this hook is running two different algorithms for sites and commands.